src/libs/sys-util-rs/src/fs.rs

```diff
@@ -57,6 +59,111 @@ pub fn is_symlink<P: AsRef<Path>>(path: P) -> std::io::Result<bool> {
     Ok(meta.file_type().is_symlink())
 }
 
+/// Reflink copy src to dst, and falls back to regular copy if reflink copy fails.
+///
+/// # Safety
+/// The `reflink_copy()` doesn't preserve permission/security context for the copied file,
+/// so caller needs to take care of it.
+pub fn reflink_copy<S: AsRef<Path>, D: AsRef<Path>>(src: S, dst: D) -> Result<()> {
+    let src_path = src.as_ref();
+    let dst_path = dst.as_ref();
+    let src = src_path.to_string_lossy();
+    let dst = dst_path.to_string_lossy();
+
+    if !src_path.is_file() {
+        return Err(eother!("reflink_copy src {} is not a regular file", src));
+    }
+
+    // Make sure dst's parent exist. If dst is a regular file, then unlink it for later copy.
+    if dst_path.exists() {
+        if !dst_path.is_file() {
+            return Err(eother!("reflink_copy dst {} is not a regular file", dst));
+        } else {
+            nix::unistd::unlink(dst_path)?;
```

Comment:
Probably better to use `std::fs::remove_file` rather than relying on nix here.

Reply:
fixed